src/components/learner-credit-management/BudgetDetailCatalogTabContents.jsx

```diff
@@ -15,7 +15,7 @@ import { configuration } from '../../config';
 const BudgetDetailCatalogTabContents = () => {
   const history = useHistory();
   const { location } = history;
-  const { state: locationState } = location;
+  const { state: locationState } = useLocation();
```

Comment:
As is, there are now different data sources for `location`. The `location` returned by `useHistory` and the `location` returned by `useLocation`. All `location` based stuff should be coming from `useLocation` now; otherwise, when `history.replace({ ...location, state: newState });` is called below the `location` used here may not be in sync with the location provided by `useLocation`.

Reply:
Yep that makes sense. Went ahead and swapped over to using the `useLocation` location object 👍🏽